src/main/scala/com/twitter/diffy/proxy/HttpDifferenceProxy.scala

```diff
@@ -34,8 +34,12 @@ trait HttpDifferenceProxy extends DifferenceProxy {
   override type Rep = HttpResponse
   override type Srv = HttpService
 
+
   override def serviceFactory(serverset: String, label: String) =
-    HttpService(Http.newClient(serverset, label).toService)
+    HttpService(
+      CloneHttpRequestFilter.apply.
+        andThen(RefineHttpHeadersByLabel(label)).
```

Comment:
I see the desire to partition the logic.  I'd prefer not to perform multiple runs through the http header set.

Let's see if we can have one filter, but aggregate the header effects.